File: src/components/intro.rs
use sycamore::prelude::*;

#[component]
pub fn Intro<G: Html>(cx: Scope) -> View<G> {
    view! {cx,
        div(class="intro") {
            h4 { "Hi 👋, My name is" }
            h1 { "Binh Tran" }
            h2 { "Full stack  web developer" }
            h2 { "Automation tester" }
            h2 { "from Vietnam" }
        }
    }
}

File: src/components/intro.ts
import { env } from "mini-van-plate/shared";

import { Social } from "./social.js";

export const IntroContent = () => {
	const { div, h1, h2, h3, span } = env.van.tags;
	return div(
		{ class: "content" },
		h3("Hi ", span({ class: "bi-waving-hand" }), ", My name is"),
		h1("BINH TRAN"),
		h2("Self Studied Developer"),
		h2("from Vietnam"),
	);
};

export const Intro = () => {
	const { div, section, canvas } = env.van.tags;
	return [
		canvas({
			class: "background",
			id: "background-hover",
			width: 1920,
			height: 1080,
		}),
		section(
			{ class: "intro container" },
			div({ id: "intro" }),
			IntroContent(),
			Social(),
		),
	];
};
